execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1933,7 +1933,7 @@ where
         &self,
         correlation_id: CorrelationId,
         get_era_validators_request: GetEraValidatorsRequest,
-    ) -> Result<Option<ValidatorWeights>, GetEraValidatorsError> {
+    ) -> Result<Option<EraValidators>, GetEraValidatorsError> {
```

Comment:
I wonder whether this should be
```rust
Result<EraValidators, GetEraValidatorsError>
```
now; because we expect `EraValidators` to always exist?

Reply:
did this @afck

Reply:
AFAIR, the reason to return `Option` here is that it's not _really_ required for `EraValidators` to exist. It's the consensus protocol that expects the results and therefore the value is optional at the EE level but not in consensus.

Reply:
Right, I guess that makes sense, too. Was just a thought.